rayon-core/src/join/mod.rs

```diff
@@ -20,6 +20,13 @@ mod test;
 /// only execute code in parallel when there are idle CPUs to handle
 /// it.
 ///
+/// When `join` is called from outside the thread pool, the calling
+/// thread will block while the closures execute in the pool.  When
+/// `join` is called within the pool, the calling thread still actively
+/// participates in the thread pool.  It will try to execute its own
+/// closures first, but if one is stolen then it will look for other
+/// work while waiting for that result to come back.
```

Comment:
Not the best example, I agree, but I think it's important to specify which will execute immediately and which is available for stealing. Maybe just delete the 2nd paragraph? Anybody who *cares* enough to be tuning like this will have all the info they need in the 1st paragraph (in that case, we could rewrite it to talk about stealing a bit, too, like "while making closure B available to be stolen"; I was trying to avoid jargon but maybe it's better this way, more precise -- we can always link to wikipedia or something).